aiida/manage/configuration/config.py

```diff
@@ -39,24 +38,30 @@ class Config(object):  # pylint: disable=too-many-public-methods
     def from_file(cls, filepath):
         """Instantiate a configuration object from the contents of a given file.
 
+        .. note:: if the filepath does not exist an empty file will be created with the default configuration.
+
         :param filepath: the absolute path to the configuration file
         :return: `Config` instance
         """
         from aiida.cmdline.utils import echo
         from .migrations import check_and_migrate_config, config_needs_migrating
 
-        with io.open(filepath, 'r', encoding='utf8') as handle:
-            config = json.load(handle)
-
-        # If the configuration file needs to be migrated, first create a specific backup so it can easily be reverted
-        if config_needs_migrating(config):
-            echo.echo_warning('current configuration file `{}` is outdated and will be migrated'.format(filepath))
-            filepath_backup = cls._backup(filepath)
-            echo.echo_warning('original backed up to `{}`'.format(filepath_backup))
+        try:
+            with io.open(filepath, 'r', encoding='utf8') as handle:
+                config = json.load(handle)
+        except (IOError, OSError):
+            config = Config(filepath, check_and_migrate_config({}))
+            config.store()
+        else:
+            # If the configuration file needs to be migrated first create a specific backup so it can easily be reverted
+            if config_needs_migrating(config):
```

Comment:
sorry, one last thing I noticed: if the config file needs migrating, will this now result in two backups?
Won't the `.store` method take care of this backup automatically?

Reply:
ah no, because there is no storing going on here...
one could potentially replace the call to `_backup` with a `.store` though..

Reply:
We could have, but for the reporting it is nice to write the filename of the created backup, which is only returned by `_backup`. So I think it is fine to keep it like this